clients/banking/src/components/CardItemPhysicalDeliveryAddressForm.tsx

```diff
@@ -0,0 +1,199 @@
+import { Future, Option } from "@swan-io/boxed";
+import { LakeLabel } from "@swan-io/lake/src/components/LakeLabel";
+import { LakeTextInput } from "@swan-io/lake/src/components/LakeTextInput";
+import { nullishOrEmptyToUndefined } from "@swan-io/lake/src/utils/nullish";
+import { pick } from "@swan-io/lake/src/utils/object";
+import { CountryPicker } from "@swan-io/shared-business/src/components/CountryPicker";
+import { PlacekitAddressSearchInput } from "@swan-io/shared-business/src/components/PlacekitAddressSearchInput";
+import { CountryCCA3, allCountries } from "@swan-io/shared-business/src/constants/countries";
+import { combineValidators, useForm } from "@swan-io/use-form";
+import { forwardRef, useImperativeHandle, useState } from "react";
+import { CompleteAddressInput } from "../graphql/partner";
+import { locale, t } from "../utils/i18n";
+import { validateAddressLine, validateRequired } from "../utils/validations";
+
+export type CardItemPhysicalDeliveryAddressFormRef = {
+  submit: () => void;
+};
+
+export type Address = {
+  addressLine1?: string | null | undefined;
+  addressLine2?: string | null | undefined;
+  city?: string | null | undefined;
+  country?: string | null | undefined;
+  postalCode?: string | null | undefined;
+  state?: string | null | undefined;
+};
+
+type Props = {
+  initialEditorState?: Address;
+  onSubmit: (editorState: CompleteAddressInput) => Future<unknown>;
+};
+
+export const CardItemPhysicalDeliveryAddressForm = forwardRef<
+  CardItemPhysicalDeliveryAddressFormRef,
+  Props
+>(({ initialEditorState, onSubmit }, ref) => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { Field, FieldsListener, setFieldValue, submitForm } = useForm({
+    addressLine1: {
+      initialValue: initialEditorState?.addressLine1 ?? "",
+      validate: combineValidators(validateRequired, validateAddressLine),
+    },
+    addressLine2: {
+      initialValue: initialEditorState?.addressLine2 ?? "",
+    },
+    postalCode: {
+      initialValue: initialEditorState?.postalCode ?? "",
+      validate: validateRequired,
+    },
+    city: {
+      initialValue: initialEditorState?.city ?? "",
+      validate: validateRequired,
+    },
+    country: {
+      initialValue: (initialEditorState?.country as CountryCCA3) ?? "FRA",
```

Comment:
Instead of casting, you can use:

```ts
match(initialEditorState?.country)
  .returnType<CountryCCA3>()
  .with(P.when(isCountryCCA3), identity)
  .otherwise(() => "FRA");
```

Reply:
fixed